pkg/shares/split_compact_shares_test.go

```diff
@@ -18,8 +18,11 @@ func TestCount(t *testing.T) {
 		{transactions: []coretypes.Tx{}, wantShareCount: 0},
 		{transactions: []coretypes.Tx{[]byte{0}}, wantShareCount: 1},
 		{transactions: []coretypes.Tx{bytes.Repeat([]byte{0}, 100)}, wantShareCount: 1},
-		{transactions: []coretypes.Tx{bytes.Repeat([]byte{0}, appconsts.ContinuationCompactShareContentSize+1)}, wantShareCount: 2},
-		{transactions: []coretypes.Tx{bytes.Repeat([]byte{0}, appconsts.ContinuationCompactShareContentSize*2+1)}, wantShareCount: 3},
+		// NOTE: Each tx will require two extra bytes for the length prefix hence why we use -2 and -1
+		{transactions: []coretypes.Tx{bytes.Repeat([]byte{0}, appconsts.FirstCompactShareContentSize-2)}, wantShareCount: 1},
```

Comment:
[optional] since the `-2` is dependent on the # of bytes used by the varint, proposal to extract something like

```go
func TestCount(t *testing.T) {
	type testCase struct {
		transactions   []coretypes.Tx
		wantShareCount int
	}
	testCases := []testCase{
		{transactions: []coretypes.Tx{}, wantShareCount: 0},
		{transactions: []coretypes.Tx{[]byte{0}}, wantShareCount: 1},
		{transactions: []coretypes.Tx{bytes.Repeat([]byte{0}, 100)}, wantShareCount: 1},
		{transactions: []coretypes.Tx{generateTx(1)}, wantShareCount: 1},
		{transactions: []coretypes.Tx{generateTx(2)}, wantShareCount: 2},
		{transactions: []coretypes.Tx{generateTx(5)}, wantShareCount: 5},
	}
	for _, tc := range testCases {
		css := NewCompactShareSplitter(appconsts.TxNamespaceID, appconsts.ShareVersionZero)
		for _, transaction := range tc.transactions {
			css.WriteTx(transaction)
		}
		got := css.Count()
		if got != tc.wantShareCount {
			t.Errorf("count got %d want %d", got, tc.wantShareCount)
		}
	}
}

// generateTx generates a transaction that occupies exactly numShares number of
// shares.
func generateTx(numShares int) coretypes.Tx {
	if numShares == 0 {
		return coretypes.Tx{}
	}
	if numShares == 1 {
		return bytes.Repeat([]byte{0}, rawTxSize(appconsts.FirstCompactShareContentSize))
	}
	return bytes.Repeat([]byte{0}, rawTxSize(appconsts.FirstCompactShareContentSize+(numShares-1)*appconsts.ContinuationCompactShareContentSize))
}

// rawTxSize returns the raw tx size that can be used to construct a
// tx of desiredSize bytes. This function is useful in tests to account for
// the length delimiter that is prefixed to a tx.
func rawTxSize(desiredSize int) int {
	return desiredSize - DelimLen(uint64(desiredSize))
}
```

Reply:
Agree, although it's unlikely that protobuf will change their encoding algorithm any time soon this is far cleaner and accurate. https://github.com/celestiaorg/celestia-app/pull/1295/commits/0d04bde933721762aba3fc82b55cc0504a81cb58